benchmark/stats/curve.go

```diff
@@ -0,0 +1,180 @@
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package stats
+
+import (
+	"crypto/sha256"
+	"encoding/csv"
+	"encoding/hex"
+	"fmt"
+	"io/ioutil"
+	"math"
+	"math/rand"
+	"os"
+	"sort"
+	"strconv"
+)
+
+// payloadCurveRange represents a line within a payload curve CSV file.
+type payloadCurveRange struct {
+	from, to int32
+	weight   float64
+}
+
+// newPayloadCurveRange receives a line from a payload curve CSV file and
+// returns a *payloadCurveRange if the values are acceptable.
+func newPayloadCurveRange(line []string) (*payloadCurveRange, error) {
+	if len(line) != 3 {
+		return nil, fmt.Errorf("invalid number of entries in line %v (expected 3)", line)
+	}
+
+	var from, to int64
```

Comment:
Should these be int32?  Wouldn't ParseInt fail if it's out of range?

Reply:
ParseInt returns a i64, so setting these to i32 would make go complain at compile time.

Reply:
Aha, right.

Actually, given that you're only parsing 32 bits, then checking if it's > `math.MaxInt32` is not necessary.  And I believe `err` will be non-nil when this happens due to out-of-range.